src/java/org/apache/cassandra/db/compression/ZstdCompressionDictionary.java

```diff
@@ -69,6 +71,29 @@ public byte[] rawDictionary()
         return rawDictionary;
     }
 
+    @Override
+    public int occupiedMemory()
+    {
+        int occupied = rawDictionary.length;
+        if (dictDecompress != null)
+        {
+            ByteBuffer buffer = dictDecompress.getByReferenceBuffer();
+            if (buffer != null)
+                occupied += buffer.capacity();
+        }
+
+        Iterator<ZstdDictCompress> iterator = zstdDictCompressPerLevel.values().iterator();
+        while (iterator.hasNext())
+        {
+            ZstdDictCompress next = iterator.next();
+            ByteBuffer byteBuffer = next.getByReferenceBuffer();
```

Comment:
I do not think the `ZstdDictCompress` and `ZstdDictDecompress` are backed by shared bytebuffer. So `getByReferenceBuffer` **will just return null**. 

I did not create the compress and decompress objects using the shared bytebuffer because `CompressionDictionary` is shared already. Maybe we can revisit the idea.